core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -164,10 +165,23 @@ public DockerClient getClient() {
     }
 
     protected DockerClient getClientForConfig(DockerClientConfig config) {
-        return DockerClientBuilder
-                    .getInstance(config)
-                    .withDockerCmdExecFactory(new TestcontainersDockerCmdExecFactory())
-                    .build();
+        DockerClientBuilder clientBuilder = DockerClientBuilder
+            .getInstance(config);
+
+        String transportType = TestcontainersConfiguration.getInstance().getTransportType();
+        if ("okhttp".equals(transportType)) {
+            clientBuilder
+                .withDockerCmdExecFactory(new OkHttpDockerCmdExecFactory());
+        } else if ("netty".equals(transportType)) {
+            clientBuilder
+                .withDockerCmdExecFactory(new TestcontainersDockerCmdExecFactory());
```

Comment:
maybe the `TestcontainersDockerCmdExecFactory` should be renamed to `NettyDockerCmdExecFactory` in one of the next major releases?

Reply:
We plan to switch to okhttp by default and keep Netty as a fallback for a few releases unless we 100% that okhttp works for everyone. Once it happen, we will delete our Netty factory completely. There is also one in `docker-java`, this is why I named it `TestcontainersDockerCmdExecFactory` back then